my-express-app/server.js

```diff
@@ -52,46 +52,87 @@ app.get("/videos", async (req, res) => {
   }
 });
 
-
+const difficultyLevels = ["Beginner", "Intermediate", "Hard"];
+const ageGroupLevels = ["5-7", "7-9", "9-11"];
 
 app.get("/recommendations", async (req, res) => {
   const selectedDifficulty = req.query.Difficulty;
   const selectedAgeGroup = req.query.AgeGroup;
   let selectedInterests = req.query.Interests;
 
-  // Check if multiple interests are selected, split them into an array if true
-  if (selectedInterests.includes(',')) {
-    selectedInterests = selectedInterests.split(',').map(interest => interest.trim());
+  if (selectedInterests.includes(",")) {
+    selectedInterests = selectedInterests
+      .split(",")
+      .map((interest) => interest.trim());
   } else {
     selectedInterests = [selectedInterests];
   }
 
   try {
-    const videos = await Videos.findAll({
-      where: {
-        Difficulty: selectedDifficulty,
-        AgeGroup: selectedAgeGroup,
-      },
-    });
+    const videos = await Videos.findAll();
+
+    let scoredVideos = videos.map((video) => {
+      const videoInterests = video.Interest.split(",").map((interest) =>
+        interest.trim(),
+      );
+
+      let interestScore = videoInterests.reduce((score, interest) => {
+        return score + (selectedInterests.includes(interest) ? 1 : 0);
+      }, 0);
+
+      let difficultyScore;
+      if (video.Difficulty === selectedDifficulty) {
+        difficultyScore = 10;
+      } else {
+        let levelDiff = Math.abs(
+          difficultyLevels.indexOf(video.Difficulty) -
+            difficultyLevels.indexOf(selectedDifficulty),
+        );
+        if (levelDiff === 1) {
+          difficultyScore = 8;
+        } else if (levelDiff === 2) {
+          difficultyScore = 5;
+        }
+      }
 
-    // Filter videos by interest manually
-    const filteredVideos = videos.filter(video => {
-      // Split the video's interests into an array
-      const videoInterests = video.Interest.split(',').map(interest => interest.trim());
-      
-      // Check if any of the video's interests are in the selected interests
-      return videoInterests.some(interest => selectedInterests.includes(interest));
+      let ageGroupScore;
+      if (video.AgeGroup === selectedAgeGroup) {
+        ageGroupScore = 10;
+      } else {
+        let levelDiff = Math.abs(
+          ageGroupLevels.indexOf(video.AgeGroup) -
+            ageGroupLevels.indexOf(selectedAgeGroup),
+        );
+        if (levelDiff === 1) {
+          ageGroupScore = 8;
+        } else if (levelDiff === 2) {
+          ageGroupScore = 5;
+        }
+      }
+
+      let totalScore = difficultyScore + interestScore + ageGroupScore;
```

Comment:
Overall, since the max score a video can get for age is 10 and for difficulty is 10 but for interests is 1. What is the rational for scoring age and difficulty for "heavily" than interests? In regular terms, why are you valuing difficulty and age more than interests in your algorithm? I would tweak the values to make the algorithm make more sense.

Also why do you give for age and difficulty possible scores of 5, 8, and 10? Why not 1, 2, 3? Why not 0, 2, 4? And especially why 5, 8, and 10 when the highest score for interest can only be 1?